### PR TITLE
added a filter reset button

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>
* Resolves: #1816 + #1817 
* Target version: master 

grafik

### CODE REVIEW

**Comment by juliushaertl** [comment]
I think this needs some design improvement as it does not seem obvious to me that this will reset the filter. It looks more like a close action for the popover. Any idea how we can make that clearer? 

cc @nextcloud/designers @Nyco @Clementine46

**Comment by raimund-schluessler** [comment]
Maybe a button at the bottom, labeled `Reset filters`.

**Comment by jakobroehrl** [comment]
Added a filter Button at the end

File: src/store/main.js [hunk 1]
```diff
@@ -256,7 +256,9 @@ export default new Vuex.Store({
 		setFilter({ commit }, filter) {
 			commit('setFilter', filter)
 		},
-		async loadBoardById({ commit }, boardId) {
+		async loadBoardById({ commit, dispatch }, boardId) {
+			const filterReset = { tags: [], users: [], due: '' }
+			dispatch('setFilter', { ...filterReset })
```
Comment (juliushaertl) [inline src/store/main.js:261]:
```suggestion
			dispatch('setFilter', filterReset)
```

File: src/components/Controls.vue [hunk 2]
```diff
@@ -265,6 +275,11 @@ export default {
 				this.$router.push({ name: 'board.details' })
 			}
 		},
+		clearfilter() {
+			const filterReset = { tags: [], users: [], due: '' }
+			this.$store.dispatch('setFilter', { ...filterReset })
```
Comment (juliushaertl) [inline src/components/Controls.vue:280]:
```suggestion
			dispatch('setFilter', filterReset)
```

File: src/components/Controls.vue [hunk 3]
```diff
@@ -265,6 +275,11 @@ export default {
 				this.$router.push({ name: 'board.details' })
 			}
 		},
+		clearfilter() {
```
Comment (juliushaertl) [inline src/components/Controls.vue:278]:
```suggestion
		clearFilter() {
```

File: src/components/Controls.vue [hunk 4]
```diff
@@ -145,6 +145,10 @@
 									@click="beforeSetFilter">
 								<label for="noDue">{{ t('deck', 'No due date') }}</label>
 							</div>
+
+							<Button :disabled="!isFilterActive" @click="clearfilter">
```
Comment (juliushaertl) [inline src/components/Controls.vue:149]:
```suggestion
							<Button :disabled="!isFilterActive" @click="clearFilter">
```

File: src/components/Controls.vue (hunk 2 above)
Comment (jakobroehrl) [reply src/components/Controls.vue:280]: why?
dispatch is not defined

Comment (jakobroehrl) [reply src/components/Controls.vue:280]: It's not working without ...

Comment (juliushaertl) [reply src/components/Controls.vue:280]: ah sorry
i was just refering to the resetFilter part ;)

**Pull request merged**